src/Consumers/ConsumerBuilder.php

```diff
@@ -220,17 +261,20 @@ public function withSecurityProtocol(string $securityProtocol): self
     }
 
     /**
+     * Enable or disable consumer auto commit option.
+     *
      * @return $this
      */
-    public function withAutoCommit(): self
+    public function withAutoCommit(bool $autoCommit = true): self
     {
-        $this->autoCommit = true;
+        $this->autoCommit = $autoCommit;
```

Comment:
This will not work as you'd expect it to:

When you provide `false` here, the `Config::getConsumerOptions()` method will fall back to the value provided in config: `kafka.auto_commit`.

You can test this by not calling `withAutoCommit()` on the builder, and setting `kafka.auto_commit` to `false`.

This will yield an `RdKafka\Exception`:

```
Bool configuration property "enable.auto.commit" cannot be set to empty value
```

Work-around is to set `kafka.auto_commit` to `'false'` to disable auto-commit

Reply:
I think if we change this
```php
if ($this->autoCommit) {
    $options['enable.auto.commit'] = 'true';
}
```
to 

```php
if (!empty($this->autoCommit)) {
    $options['enable.auto.commit'] = $this->autoCommit === true ? 'true' : 'false';
}
```

it will be solved. What do you think?

Reply:
Yes, that seems a good fix. I don't have time to test it properly right now.
One thing I just noticed is that the same bool-to-string conversion needs to be done in preparing the initial `$options` array just a few lines up. If you set `kafka.auto_commit` to `false` instead of `'false'` the same exception will occur

Reply:
nice catch btw